charms/osm/libansible.py

```diff
@@ -0,0 +1,108 @@
+##
+# Copyright 2020 Canonical Ltd.
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+##
+
+import fnmatch
+import os
+import yaml
+import subprocess
+import sys
+
+sys.path.append("lib")
+import charmhelpers.fetch
+
+
+ansible_hosts_path = "/etc/ansible/hosts"
+
+
+def install_ansible_support(from_ppa=True, ppa_location="ppa:ansible/ansible"):
+    """Installs the ansible package.
+
+    By default it is installed from the `PPA`_ linked from
+    the ansible `website`_ or from a ppa specified by a charm config..
+
+    .. _PPA: https://launchpad.net/~rquillo/+archive/ansible
+    .. _website: http://docs.ansible.com/intro_installation.html#latest-releases-via-apt-ubuntu
+
+    If from_ppa is empty, you must ensure that the package is available
+    from a configured repository.
+    """
+    if from_ppa:
+        charmhelpers.fetch.add_source(ppa_location)
+        charmhelpers.fetch.apt_update(fatal=True)
+    charmhelpers.fetch.apt_install("ansible")
+    with open(ansible_hosts_path, "w+") as hosts_file:
+        hosts_file.write("localhost ansible_connection=local")
+
+
+def create_hosts(hostname, username, password, hosts):
+    inventory_path = "/etc/ansible/hosts"
+
+    with open(inventory_path, "w") as f:
+        f.write("[{}]\n".format(hosts))
+        h1 = "VyOS-Openstack ansible_host={0} ansible_user={1} ansible_password={2}\n".format(
+            hostname, username, password
+        )
+        f.write(h1)
+
+
+def create_ansible_cfg():
+    ansible_config_path = "/etc/ansible/ansible.cfg"
```

Comment:
Should there be a way of importing a different config?

Reply:
May be yes, but I'm not very familiar with the options available in the config. I know this just disables the host_key_chacking so that the ansible playbook can be executed without any human interaction.